certification/internal/policy/container/base_on_ubi.go

```diff
@@ -7,13 +7,24 @@ import (
 
 	cranev1 "github.com/google/go-containerregistry/pkg/v1"
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	pyxis "github.com/redhat-openshift-ecosystem/openshift-preflight/certification/pyxis"
 	log "github.com/sirupsen/logrus"
 )
 
 // BasedOnUBICheck evaluates if the provided image is based on the Red Hat Universal Base Image
 // by inspecting the contents of the `/etc/os-release` and identifying if the ID is `rhel` and the
 // Name value is `Red Hat Enterprise Linux`
-type BasedOnUBICheck struct{}
+type BasedOnUBICheck struct {
+	LayerHashCheckEngine layerHashChecker
+}
+
+type layerHashChecker interface {
+	CheckRedHatLayers(layerHashes []cranev1.Hash) ([]pyxis.CertImage, error)
+}
+
+func NewBasedOnUbiCheck(layerHashChecker layerHashChecker) *BasedOnUBICheck {
```

Comment:
When this is called you can just pass a PyxisEngine, since it satisfies the layerHashChecker interface.